File: pkg/env/env.go
package env

import (
	"fmt"
	"log"
	"os"
	"strconv"

	"github.com/joho/godotenv"
)

// Load loads environment variables from .env file
func Load() {
	if err := godotenv.Load(); err != nil {
		log.Printf("No .env file found")
	}
}

// RequiredStringVariable returns the value of an environment variable or panics if not set
func RequiredStringVariable(name string) string {
	value := os.Getenv(name)
	if value == "" {
		panic(fmt.Sprintf("required environment variable %s is not set", name))
	}
	return value
}

// RequiredIntVariable returns the value of an environment variable as int or panics if not set
func RequiredIntVariable(name string) int {
	value := RequiredStringVariable(name)
	intValue, err := strconv.Atoi(value)
	if err != nil {
		panic(fmt.Sprintf("environment variable %s must be an integer, got: %s", name, value))
	}
	return intValue
}

// StringVariable returns the value of an environment variable or a default value
func StringVariable(name, defaultValue string) string {
	if value := os.Getenv(name); value != "" {
		return value
	}
	return defaultValue
}

File: src/main/java/com/visionex/config/Env.java
package com.visionex.config;

import io.github.cdimascio.dotenv.Dotenv;

public final class Env {
    private static final Dotenv DOTENV = Dotenv.configure().ignoreIfMissing().load();

    private Env() {}

    public static String required(String name) {
        String value = get(name);
        if (value == null || value.isEmpty()) {
            throw new IllegalStateException("required environment variable " + name + " is not set");
        }
        return value;
    }

    public static int requiredInt(String name) {
        String value = required(name);
        try {
            return Integer.parseInt(value);
        } catch (NumberFormatException ex) {
            throw new IllegalStateException("environment variable " + name + " must be an integer, got: " + value, ex);
        }
    }

    public static String optional(String name, String defaultValue) {
        String value = get(name);
        return value == null || value.isEmpty() ? defaultValue : value;
    }

    public static String get(String name) {
        String value = System.getenv(name);
        if (value != null && !value.isEmpty()) {
            return value;
        }
        return DOTENV.get(name);
    }
}

